kafka/consumer.go

```diff
@@ -156,6 +172,7 @@ func (c *Consumer) Close() error {
 	// Since we take the full lock when closing, there's no need to explicitly
 	// allow rebalances since polls aren't concurrent with Close().
 	c.client.Close()
+	c.consumer.close() // Close each partition consumer.
```

Comment:
does `Client.Close` not invoke the partition revoked callbacks?

Reply:
> ... but we don't do need that, because we're explicitly committing even for at-least-once?

Correct, for at-least-once, each goroutine is responsible to commit the processed offsets. For at-most-once, we commit ahead of time, so again there's no need to commit on the revoked callback.

> does Client.Close not invoke the partition revoked callbacks?

You're right, I missed this bit:

> Internally, this function leaves the group, and leaving a group causes a rebalance so that you can get one final notification of revoked partitions

I can remove the `c.consumer.close()` call since it's redundant.